### PR TITLE
Fix decoder cancellation tests

### PR DESCRIPTION
**Pull request by antonfirsov**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

This is a follow-up on https://github.com/SixLabors/ImageSharp/pull/2276#discussion_r1032733686. Doing PR'ception for easier tracking of the changes. May push some tests for `WithSeekableMemoryStreamAsync` later, since it's needed for the logical completeness of the functional coverage trying to achieve here.

### CODE REVIEW

**Comment by antonfirsov** [comment]
Damn, no test runs here.

**Comment by JimBobSquarePants** [comment]
>Damn, no test runs here.

Let's merge it into the PR branch and continue there then. The changes look good to me.

File: tests/ImageSharp.Tests/Image/ImageTests.Decode_Cancellation.cs [hunk 1]
```diff
@@ -16,65 +16,105 @@ public class Decode_Cancellation : ImageLoadTestBase
 
         public Decode_Cancellation() => this.TopLevelConfiguration.StreamProcessingBufferSize = 128;
 
-        public static readonly TheoryData<string> TestFiles = new()
+        private static TheoryData<bool, string, double> GetTestData()
```
Comment (JimBobSquarePants) [inline tests/ImageSharp.Tests/Image/ImageTests.Decode_Cancellation.cs:19]: It's never occured to me to generate theory data in this manner. Big chore saver!

**Review by JimBobSquarePants — APPROVED** [review]

**Comment by antonfirsov** [comment]
I just pushed a fix in 00188456b1fefd3fd6c66256913529c77f431e28 😆

**Comment by antonfirsov** [comment]
Should be good now in #2276.

**Pull request merged**